### ISSUE TITLE
Sqlite TypeError - must be real number, not str

### ISSUE THREAD

**Issue by ahmedsaud-01** [Farzad-R/Advanced-RAG-Series#7]
Hi @Farzad-R ,

I tried to execute notebook AgentGraph-Intelligent-Q&A-and-RAG-System/Notebooks/Tools/sql_agents/sql_agent_chain_steps.ipynb. But I am getting this error in Line 18.  Link to  github file: [https://github.com/Farzad-R/Advanced-QA-and-RAG-Series/blob/main/AgentGraph-Intelligent-Q%26A-and-RAG-System/Notebooks/Tools/sql_agents/sql_agent_chain_steps.ipynb](https://github.com/Farzad-R/Advanced-QA-and-RAG-Series/blob/main/AgentGraph-Intelligent-Q%26A-and-RAG-System/Notebooks/Tools/sql_agents/sql_agent_chain_steps.ipynb)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], [line 2](vscode-notebook-cell:?execution_count=6&line=2)
      [1](vscode-notebook-cell:?execution_count=6&line=1) message = "How many tables do I have in the database? and what are their names?"
----> [2](vscode-notebook-cell:?execution_count=6&line=2) response = chain.invoke({"question": message})
      [3](vscode-notebook-cell:?execution_count=6&line=3) response

File c:\Users\saud_\source\repos\Advanced-QA-and-RAG-Series\AgentGraph-Intelligent-Q&A-and-RAG-System\abc\Lib\site-packages\langchain_core\runnables\base.py:3022, in RunnableSequence.invoke(self, input, config, **kwargs)
   [3020](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/base.py:3020) context.run(_set_config_context, config)
   [3021](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/base.py:3021) if i == 0:
-> [3022](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/base.py:3022)     input = context.run(step.invoke, input, config, **kwargs)
   [3023](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/base.py:3023) else:
   [3024](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/base.py:3024)     input = context.run(step.invoke, input, config)

File c:\Users\saud_\source\repos\Advanced-QA-and-RAG-Series\AgentGraph-Intelligent-Q&A-and-RAG-System\abc\Lib\site-packages\langchain_core\runnables\passthrough.py:494, in RunnableAssign.invoke(self, input, config, **kwargs)
    [488](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:488) def invoke(
    [489](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:489)     self,
    [490](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:490)     input: dict[str, Any],
    [491](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:491)     config: Optional[RunnableConfig] = None,
    [492](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:492)     **kwargs: Any,
    [493](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:493) ) -> dict[str, Any]:
--> [494](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/passthrough.py:494)     return self._call_with_config(self._invoke, input, config, **kwargs)

File c:\Users\saud_\source\repos\Advanced-QA-and-RAG-Series\AgentGraph-Intelligent-Q&A-and-RAG-System\abc\Lib\site-packages\langchain_core\runnables\base.py:1927, in Runnable._call_with_config(self, func, input, config, run_type, serialized, **kwargs)
   [1923](file:///C:/Users/saud_/source/repos/Advanced-QA-and-RAG-Series/AgentGraph-Intelligent-Q&A-and-RAG-System/abc/Lib/site-packages/langchain_core/runnables/base.py:1923)     context = copy_context()
...
File lib\\sqlalchemy\\cyextension\\resultproxy.pyx:79, in sqlalchemy.cyextension.resultproxy._apply_processors()

File lib\\sqlalchemy\\cyextension\\processors.pyx:68, in sqlalchemy.cyextension.processors.DecimalResultProcessor.process()

TypeError: must be real number, not str
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?1e2b2354-c444-451b-b394-945890c354ad) or open in a [text editor](command:workbench.action.openLargeOutput?1e2b2354-c444-451b-b394-945890c354ad). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

**Comment by Farzad-R** [comment]
Hello ahmedsaud,

Can you check the version of your Langchain, LangGraph, and OpenAI libraries? please share them with me here

**Comment by ahmedsaud-01** [comment]
Hello @Farzad-R,
Thank you for your reply

My python version is  3.12.7 and I have these versions of Langchain, LangGraph and OpenAI libraries.
langchain==0.2.16
langgraph==0.2.19
openai==1.44.0

**Comment by Farzad-R** [comment]
Thank you. Everything looks good except for the Python version. I used 3.11. So, give it a try using python 3.11 and see if that would fix the problem.

Unfortunately, the error traceback isn't clear enough to pinpoint the exact cause. So if that didn't solve the problem, here’s what I suggest: have you tried running the notebooks for the SQL agents? The notebooks include a step-by-step implementation, which should make debugging easier. Please run the code in the notebook and let me know which cell is causing the error, along with the error message.

**Comment by ahmedsaud-01** [comment]
Thank you Farzad,
I have executed the notebook name "sql_agent_chain_for_large_db.ipynb" from start to end and it didn't give me errors.
But when I execute notebook "sql_agent_chain_steps.ipynb" it gives me error in question  "How many tables do I have in the database? and what are their names?" but If I change the question and ask only the name of tables it works without throwing any error. python version 3.11 gives me few error as I noticed

![image](https://github.com/user-attachments/assets/bab51395-b638-4152-9fe3-7a8a5ff9c204)


Thank you again
